src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -142,9 +142,9 @@ bool IsValidTriggerCharacterForDocument(Document document, char triggerCharacter
                 var completionItem = new TCompletionItem
                 {
                     Label = completeDisplayText,
-                    InsertText = item.Properties.ContainsKey("InsertionText") ? item.Properties["InsertionText"] : completeDisplayText,
-                    SortText = item.SortText,
-                    FilterText = item.FilterText,
+                    InsertText = (item.Properties.ContainsKey("InsertionText") && !item.Properties["InsertionText"].Equals(item.DisplayText, StringComparison.Ordinal)) ? item.Properties["InsertionText"] : null,
+                    SortText = (item.HasDifferentSortText) ? item.SortText : null,
```

Comment:
I believe this also needs to be comparing against the label since that is the fallback in LSP.  You may be able to get rid of the new property on the underlying roslyn completion item as well

Reply:
One of the two `HasDifferent` properties already existed, but the implementation compares against the base display name. From an LSP perspective it makes sense to compare against the full label, but I wonder about the usage of `SortText` and `FilterText` as used from `CompletionItem` since they default to a different value than LSP would.

Reply:
can LSP just compress these streams instead?  taht seems like it would get a ton of bang/buck as then the compression logic will see and dedupe tehse things.  then we dont' have to manually try to make json efficient here.

Reply:
Compression wouldn't solve the serialize/deserialize time which is the primary driver behind this change.

Reply:
>but I wonder about the usage of SortText and FilterText as used from CompletionItem since they default to a different value than LSP would.

I think as long as we only omit the sort text (and others) as long as it is exactly the same as the LSP label, then we won't be changing current LSP behavior, since we're only modifying the LSP payload according to other data in the LSP payload

Reply:
> I think as long as we only omit the sort text (and others) as long as it is exactly the same as the LSP label, then we won't be changing current LSP behavior, since we're only modifying the LSP payload according to other data in the LSP payload

I think you're right, unfortunately I think it means that we will still always specify these fields in your IntelliCode example and any other usage of prefix/postfix. Perhaps moving flags that would indicate prefix/postfixes to `data` might be in order outside this PR.

Reply:
>unfortunately I think it means that we will still always specify these fields in your IntelliCode example and any other usage of prefix/postfix

I'm pretty sure these are not very common.  E.g. for intellicode only a few of the items will have it.  

>Perhaps moving flags that would indicate prefix/postfixes to data might be in order outside this PR.
Yup, no need to do here.